drivers/sensor/lps28dfw/lps28dfw.c

```diff
@@ -0,0 +1,332 @@
+		LOG_DBG("Failed to read sample");
+		return -EIO;
+	}
+
+	data->sample_press = raw_data.pressure.raw;
+	data->sample_temp = raw_data.heat.raw;
+
+	return 0;
+}
+
+static inline void lps28dfw_press_convert(const struct device *dev,
+					 struct sensor_value *val,
+					 int32_t raw_val)
+{
+	int32_t press_tmp = raw_val >> 8; /* raw value is left aligned (24 msb) */
+	lps28dfw_md_t md;
+
+	md.fs = ((struct lps28dfw_config *)dev->config)->fs;
+
+	/* Pressure sensitivity is:
+	 * - 4096 LSB/hPa for Full-Scale of 260 - 1260 hPa:
+	 * - 2048 LSB/hPa for Full-Scale of 260 - 4060 hPa:
+	 * Also convert hPa into kPa
+	 */
+	if (md.fs == 0) {
+		val->val1 = press_tmp / 40960;
+	} else if (md.fs == 1) {
+		val->val1 = press_tmp / 20480;
+	}
+
+	/* For the decimal part use (3125 / 128) as a factor instead of
+	 * (1000000 / 40960) to avoid int32 overflow
+	 */
+	val->val2 = (press_tmp % 40960) * 3125 / 128;
+}
+
+static inline void lps28dfw_temp_convert(struct sensor_value *val,
+					int16_t raw_val)
+{
+	/* Temperature sensitivity is 100 LSB/deg C */
+	val->val1 = raw_val / 100;
+	val->val2 = ((int32_t)raw_val % 100) * 10000;
+}
+
+static int lps28dfw_channel_get(const struct device *dev,
+			       enum sensor_channel chan,
+			       struct sensor_value *val)
+{
+	struct lps28dfw_data *data = dev->data;
+
+	if (chan == SENSOR_CHAN_PRESS) {
+		lps28dfw_press_convert(dev, val, data->sample_press);
+	} else if (chan == SENSOR_CHAN_AMBIENT_TEMP) {
+		lps28dfw_temp_convert(val, data->sample_temp);
+	} else {
+		return -ENOTSUP;
+	}
+
+	return 0;
+}
+
+static const uint16_t lps28dfw_map[] = {0, 1, 4, 10, 25, 50, 75, 100, 200};
+
+static int lps28dfw_odr_set(const struct device *dev, uint16_t freq)
+{
+	int odr;
+
+	for (odr = 0; odr < ARRAY_SIZE(lps28dfw_map); odr++) {
+		if (freq == lps28dfw_map[odr]) {
+			break;
+		}
+	}
+
+	if (odr == ARRAY_SIZE(lps28dfw_map)) {
+		LOG_DBG("bad frequency");
+		return -EINVAL;
+	}
+
+	if (lps28dfw_set_odr_raw(dev, odr) < 0) {
+		LOG_DBG("failed to set sampling rate");
+		return -EIO;
+	}
+
+	return 0;
+}
+
+static int lps28dfw_attr_set(const struct device *dev,
+			    enum sensor_channel chan,
+			    enum sensor_attribute attr,
+			    const struct sensor_value *val)
+{
+	if (chan != SENSOR_CHAN_ALL) {
+		LOG_WRN("attr_set() not supported on this channel.");
+		return -ENOTSUP;
+	}
+
+	switch (attr) {
+	case SENSOR_ATTR_SAMPLING_FREQUENCY:
+		return lps28dfw_odr_set(dev, val->val1);
+	default:
+		LOG_DBG("operation not supported.");
+		return -ENOTSUP;
+	}
+
+	return 0;
+}
+
+static const struct sensor_driver_api lps28dfw_driver_api = {
+	.attr_set = lps28dfw_attr_set,
+	.sample_fetch = lps28dfw_sample_fetch,
+	.channel_get = lps28dfw_channel_get,
+#if CONFIG_LPS28DFW_TRIGGER
+	.trigger_set = lps28dfw_trigger_set,
+#endif
+};
+
+static int lps28dfw_init_chip(const struct device *dev)
+{
+	const struct lps28dfw_config * const cfg = dev->config;
+	__maybe_unused struct lps22df_data *data = dev->data;
```

Comment:
can you move this in the `if()` block where it's used? then you can do without `__maybe_unused`